documentation/course/e2/README.md

```diff
@@ -2069,15 +2074,55 @@ TODO @jakubzehner: Dodać diagram pakietów, opis architektury i endpointy.
-| `GET`      | `/int/v1/health` | clabbert      | —             | Sprawdzenie stanu głównego serwisu ([`M/03`](#m03-healthchecki-dla-serwisów)). |
+| `GET`      | `/int/v1/health` | Clabbert      | —             | Sprawdzenie stanu głównego serwisu ([`M/03`](#m03-healthchecki-dla-serwisów)). |
 
 ## Płatność
 
 TODO @piterek130: Dodać diagram pakietów, opis architektury i endpointy.
 
 ## Logistyka
 
-TODO @mlodybercik: Dodać diagram pakietów, opis architektury i endpointy.
+![Diagram pakietów Leprechaun](./images/package-diagram-leprechaun.drawio.svg)
+
+Część systemu obsługująca logistykę JakPrzyjade, zarządzająca trasami, przystankami, pojazdami, awariami oraz rozkładami jazdy podzielona została na trzy pakiety realizujące inne zadania w działaniu aplikacji obsługującej REST API. Każde z nich jest odpowiedzialne za integralną część działania tego podsystemu i są to pakiety odpowiedzialne za komunikację (`controllers`) odpowiadające na zapytania do systemu, za logikę biznesową (`services`), oraz za dostęp do bazy danych oraz mapowanie danych przychodzących na obiekty bazodanowe oraz odwrotnie (`data`). Jednym z dodatkowych wydzielonych pakietów jest pakiet organizujący funkcje pomocnicze (`helpers`) gotowe do ponownego użycia w innych częściach systemu.
+
+### API
+
+#### API publiczne
+
+| **Rola**                                    | **Metoda** | **Endpoint**              | **Wymagania**                          | **Opis**                                       |
+| ------------------------------------------- | ---------- | ------------------------- | -------------------------------------- | ---------------------------------------------- |
+| `passenger`, `admin`                        | `GET`      | `/ext/v1/lines`           | `LOG/01`, `LOG/02`                     | Pobranie listy aktywnych linii.                |
+| `passenger`, `admin`                        | `GET`      | `/ext/v1/lines/:id`       | `LOG/01`, `LOG/02`                     | Pobranie informacji o linii.                   |
+| `admin`                                     | `POST`     | `/ext/v1/lines`           | `LOG/03`                               | Utworzenie nowej linii.                        |
+| `admin`                                     | `PATCH`    | `/ext/v1/lines/:id`       | `LOG/04`                               | Edytowanie linii.                              |
+| `admin`                                     | `DELETE`   | `/ext/v1/lines/:id`       | `LOG/05`                               | Dezaktywacja linii.                            |
+| `passenger`, `admin`                        | `GET`      | `/ext/v1/lines/:id/stops` | `LOG/06`, `LOG/07`                     | Pobranie listy przystanków konkretnej linii.   |
+| `passenger`, `admin`                        | `GET`      | `/ext/v1/stops/:id/lines` | `LOG/06`, `LOG/07`                     | Pobranie listy linii obsługujących przystanek. |
```

Comment:
**issue:** Getting a list of all stops (better yet with name filtering) seems to be missing, yet useful, e.g. here:
![obraz](https://github.com/user-attachments/assets/5ebdabdb-ca85-4bf0-8424-fd8cb515c34b)